src/main/java/Application/OpenCVServer/OpenCVServerOperator.java

```diff
@@ -0,0 +1,141 @@
+package Application.OpenCVServer;
+
+import java.io.*;
+import java.net.*;
+
+import javax.imageio.ImageIO;
+import java.awt.image.BufferedImage;
+import java.util.ArrayList;
+import java.util.List;
+
+import Application.Utilities.DetectMarker;
+import Application.Utilities.HeartBeatRunner;
+import Application.Utilities.ImageProcessor;
+import Application.Utilities.OpenCVUtil;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import org.opencv.core.Mat;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+
+public class OpenCVServerOperator {
+    private ServerSocket serverSocket;
+    private Socket clientSocket;
+    private DataInputStream in;
+    private PrintWriter out;
+    private DetectMarker detector;
+    private int port;
+
+    private String fileName;
+
+    private BufferedImage currentImage;
+
+    public ServerRunner serverRunner;
+    public HeartBeatRunner beatRunner;
+
+    private static OpenCVServerOperator instance = null;
+
+    private OpenCVServerOperator() {
+        fileName= "read.png";
+        detector = new DetectMarker();
+    }
+
+    public void startConnection() throws IOException {
+        serverSocket = new ServerSocket(port);
+        clientSocket = serverSocket.accept();
+        out = new PrintWriter(clientSocket.getOutputStream(), true);
+        in = new DataInputStream(clientSocket.getInputStream());
+    }
+
+    // change name to something more appropriate
+    public void standbyForConnection() throws IllegalArgumentException, IOException {
+        //if input is faulty, timeout and continue
```

Comment:
I don't understand that.

Reply:
it is an old TODO , for now it has been implemented by the Server throwing an IOException and continuing the loop